junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/TestMethodTestDescriptor.java

```diff
@@ -230,4 +234,48 @@ private void invokeAfterEachCallbacks(JupiterEngineExecutionContext context) {
 		});
 	}
 
+	@Override
+	public void nodeSkipped(JupiterEngineExecutionContext context, TestDescriptor descriptor, SkipResult result) {
+		invokeTestWatchers(context, t -> {
```

Comment:
Please rework the implementation so that watchers are invoked in _registration order_ for skipped events and otherwise in _reverse registration order_.